client/my-sites/checklist/wpcom-checklist/checklist-notification/index.jsx

```diff
@@ -15,31 +14,16 @@ export class ChecklistNotification extends Component {
 	}
 
 	shouldShowNotification = () => {
-		const { storedTask, canShowChecklist, children } = this.props;
-
-		// The redux state for getSiteChecklist() is injected from an API response
-		// so the selector's results and the task list will not always match
-		// Therefore, we're accessing the child Tasks directly for accuracy
-
-		const childrenArray = Children.toArray( children );
-		const task = find( childrenArray, child => ! child.props.completed );
-		const total = childrenArray.length;
-		const completeCount = reduce(
-			childrenArray,
-			( sum, child ) => ( !! child.props.completed ? sum + 1 : sum ),
-			0
-		);
-		const isFinished = completeCount >= total;
-
-		this.props.setNotification( false );
+		const { storedTask, canShowChecklist, taskList } = this.props;
 
 		if (
-			task &&
-			( storedTask !== task.key || storedTask === null ) &&
-			! isFinished &&
+			! taskList.areAllTasksCompleted() &&
+			( storedTask !== taskList.getFirstIncompleteTask().id || storedTask === null ) &&
```

Comment:
Could we eliminate the check for `areAllTasksCompleted()`? That's equivalent to `getFirstIncompleteTask()` returning null, isn't it?

Reply:
It is. It was more of a conceptual thing, but you're right, "no first incomplete task" is easy enough to grasp to not require another method.